### DISCUSSION TITLE
Using VS Code to remotely program Raspberry Pi can't find header files/namespaces and Intellisense doesn't work

### DISCUSSION THREAD

**Discussion by dlptoo** [microsoft/vscode-cpptools#8107] [Q&A] [answered]
Hi! I'm not sure if this is a problem with the C/C++ extension or just with trying to use VS Code remotely to program on a Raspberry Pi.

If I use VS Code directly on the Pi as per https://code.visualstudio.com/docs/cpp/config-linux, everything works fine. The header files and namespaces are found automatically and Intellisense works fine. If I remote into the Pi as per https://www.raspberrypi.org/blog/coding-on-raspberry-pi-remotely-with-visual-studio-code/, the header files/namespaces are not found and Intellisense doesn't work. I assume the Intellisense issue is because of the header/namespace issue.

Is this a problem with the C/C++ extension, or should I be looking elsewhere to pin down the cause and find a fix?

Thanks!

**Comment by sean-mcmanus** [comment]
This should work. You may need to set some remote settings, e.g. if you have a user level C_Cpp.default.compilerPath set. You can run the C/C++: Log Diagnostics command in both scenarios and compare the difference to try to determine what is going wrong.

**Comment by dlptoo** [comment, marked as answer]
Sean,
Thanks! The problem was actually my fault. I named my source file HelloWorldRemote instead of HellowWorldRemote.cpp. Duh... Once I added the file extension, everything worked fine.

Sorry!
Don
